apm-agent-plugins/apm-spring-webmvc/apm-spring-webmvc-plugin/src/main/java/co/elastic/apm/agent/springwebmvc/Spring6ServiceNameInstrumentation.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.agent.springwebmvc;
+
+import co.elastic.apm.agent.servlet.Constants;
+import co.elastic.apm.agent.servlet.adapter.JakartaServletApiAdapter;
+import net.bytebuddy.asm.Advice;
+import org.springframework.web.context.WebApplicationContext;
+
+public class Spring6ServiceNameInstrumentation extends AbstractSpringServiceNameInstrumentation {
+
+    @Override
+    public Constants.ServletImpl servletImpl() {
+        return Constants.ServletImpl.JAKARTA;
+    }
+
+    public static class AdviceClass extends AbstractSpringServiceNameInstrumentation.AdviceClass {
```

Comment:
[minor] if we attempt to not use advice class inheritance, then maybe moving the `onExit` implementation directly to the adapter could be an option (or use an helper class). While the advice inheritance is definitely possible and I have nothing against it, it's not a pattern widely used, also we have to open the actual `onExit` implementation to know what it does, whereas a call to an helper class method usually have a more descriptive name.

Reply:
> then maybe moving the onExit implementation directly to the adapter could be an option

I would prefer to not that do that because it contains spring-specific code (the `application.name` lookup).

> While the advice inheritance is definitely possible and I have nothing against it, it's not a pattern widely used

That's true, I used our servlet-api as template here.

> whereas a call to an helper class method usually have a more descriptive name

I will switch to a static helper method here aswell.